In [23]:
import pandas as pd
import numpy as np
import math

In [25]:
data = {
    'Satisfaction': ['Very Satisfied', 'Satisfied', 'Neutral', 'Unsatisfied', 'Very Unsatisfied', 'Total'],
    'Smart Thermostat': [50, 80, 60, 30, 20, 240],
    'Smart Light': [70, 100, 90, 50, 50, 360]
}

In [26]:
df = pd.DataFrame(data)


In [27]:
df

Satisfaction  Smart Thermostat  Smart Light
0    Very Satisfied                50           70
1         Satisfied                80          100
2           Neutral                60           90
3       Unsatisfied                30           50
4  Very Unsatisfied                20           50
5             Total               240          360

In [28]:
df.set_index('Satisfaction', inplace=True)


In [58]:
df.drop(index='Total',inplace=True)

In [59]:
df

Smart Thermostat  Smart Light
Satisfaction                                   
Very Satisfied                  50           70
Satisfied                       80          100
Neutral                         60           90
Unsatisfied                     30           50
Very Unsatisfied                20           50

 Objective:
 
To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.


#### 1. State the Hypotheses:


Null Hypothesis (H0): There is no association between the type of smart home device purchased and customer satisfaction level.

Alternative Hypothesis (H1): There is an association between the type of smart home device purchased and customer satisfaction level.

#### 2. Compute the Chi-Square Statistic:


In [60]:
from scipy.stats import chi2_contingency


In [62]:
chi2, p, dof, expected = chi2_contingency(df)


In [63]:
chi2

5.638227513227513

#### 3. Determine the Critical Value:


In [64]:
## degree of freedom

In [66]:
r,c=df.shape

In [68]:
dof=(r-1)*(c-1)
dof

4

In [71]:
from scipy.stats import chi2
alpha=0.05
t_crictical=chi2.ppf(1-alpha,4)
t_crictical

9.487729036781154

#### 4. Make a Decision:


Since, 5.5564<9.488, the calculated Chi-Square statistic does not exceed the critical value.

we fail to reject the null hypothesis.

This means that there is a significant association between the type of smart home device purchased and the customer satisfaction level.